addons/pos_loyalty/models/pos_order.py

```diff
@@ -97,7 +97,7 @@ def confirm_coupon_programs(self, coupon_data):
         for old_id, new_id in zip(coupons_to_create.keys(), new_coupons):
             coupon_new_id_map[new_id.id] = old_id
 
-        all_coupons = self.env['loyalty.card'].browse(coupon_new_id_map.keys()).exists()
+        all_coupons = self.env['loyalty.card'].sudo().browse(coupon_new_id_map.keys()).exists()
```

Comment:
how come we didn't need the `sudo` before ?

Reply:
it looks like pos users have `read` and `write` permissions on the `loyalty.card` model
https://github.com/odoo/odoo/blob/b80684d67c8a153b435b2475810ba917a2d2bd7d/addons/pos_loyalty/security/ir.model.access.csv#L6

Reply:
It's happening because we don't have access to `sale.order.line` when we compute `_compute_order_count` on the loyalty program